src/lib/encoder.ts

```diff
@@ -325,6 +326,10 @@ const valueTypeEncodingMap = {
     decode: (value: string) => abiCoder.decodeParameter('address', value),
   },
   // NOTE: We could add conditional handling of numeric values here...
+  uint128: {
+    encode: (value: string | number) => padLeft(value, 32),
```

Comment:
what happens if i pass "HelloMyFriend" as an input? It will encode as:

`0x000....000HelloMyFriend` which is not hex and something will fail very hard later in the process no?

What if input is `123456789` ? Will it encode as `0x000...0000123456789` ? Shouldn't it be hex (not decimal) ??

Do we have checks that prevent having corrupted values?